docs/source/examples/Semantic_segmentation.rst

```diff
@@ -0,0 +1,246 @@
+    _mean = [0.485, 0.456, 0.406]  # mean of the imagenet dataset for normalizing 
+    _std = [0.229, 0.224, 0.225]  # std of the imagenet dataset for normalizing 
+    set_seeds(42)
+    
+Loading the VOCSegmentation dataset    
+==================================
+
+The VOCSegmentation dataset can be easily downloaded from torchvision.datasets. This dataset allows you, to apply the needed transforms on the ground-truth, directly, in addition to defining the proper transforms for the input images. To do so, it's enough to use the target_transfromargument and set it to your transform function of interest.
+
+.. code-block:: python
+
+    input_transform = tfms.Compose([
+        tfms.Resize((image_size, image_size)),  
+        tfms.ToTensor(),        
+        tfms.Normalize(_mean, _std)
+    ])
+    
+    target_transform = tfms.Compose([
+        tfms.Resize((image_size, image_size)),  
+        tfms.ToTensor(),        
+    ])
+    
+    # Creating the dataset
+    train_dataset = torchvision.datasets.VOCSegmentation('./dataset/', year='2007', download=True, image_set='train', transform=input_transform, target_transform= target_transform)
+    valid_dataset = torchvision.datasets.VOCSegmentation('./dataset/', year='2007', download=True, image_set='val', transform=input_transform, target_transform= target_transform )
+    
+    # Creating the dataloader
+    train_dataloader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)
+    valid_dataloader = DataLoader(valid_dataset, batch_size=batch_size, shuffle=False)
+
+A random batch of the VODSegmentation dataset images
+==================================
+
+Let's see some of the input samples, inside the training dataset.
+
+.. code-block:: python
+
+    samples = next(iter(train_dataloader))
+    inputs = samples[0]
+    input_grid = make_grid(inputs)
+    
+    fig = plt.figure(figsize=(10, 10))
+    input_grid = input_grid.numpy()
+    input_grid = input_grid.transpose((1, 2, 0)) * _std + _mean
+    inp = np.clip(input_grid, 0, 1)
+    plt.imshow(inp)
+    plt.axis('off')
+    plt.show()
+    
+.. image:: /_static/img/voc_segment_batch.png 
+
+The ground-truth (segmentation map) for the image grid shown above, is as below.
+
+.. code-block:: python
+
+    ground_truth = samples[1]
+    input_grid = make_grid(ground_truth)
+    inp = input_grid.numpy()[0]
+    
+    fig = plt.figure(figsize=(10, 10))
+    plt.imshow(inp)
+    plt.axis('off')
+    plt.show()
+    
+.. image:: /_static/img/voc_segment_batch_gt.png 
+
+It is worth mentioning that, as we have approached the segmentation task as an image translation problem, we took advantage of MSELoss for the training. Moreover, we believe that using the Unet, with a pre-trained encoder, would help the network converge sooner and better, since a pre-trained CNN (trained on the Imagenet dataset), is already familiar with some of the object classes, and also some low-level image features (such as edge, texture, etc).
```

Comment:
```suggestion
It is worth mentioning that, as we have approached the segmentation task as an image translation problem, we take advantage of MSELoss for the training. Moreover, we believe that using the U-Net with a pre-trained encoder would help the network converge sooner and better since a pre-trained CNN (trained on the ImageNet dataset) is already familiar with some of the object classes and also some low-level image features (such as edge, texture, etc).
```